### ISSUE TITLE
Random Ores Showing Up For My Players When Not Using Any Xray

### ISSUE THREAD

**Issue by Noobymen** [Imprex-Development/orebfuscator#261]
### Description

When my player are just mining stuff random ores just show up for them on the ground out of no where

![image](https://user-images.githubusercontent.com/98334209/211606523-4adc10a0-2319-4e99-ad2a-2acf3a65e897.png)

### Reproduction Steps

using paper minecraft server version 1.19.3 

### Expected Behaviour

ores to not show up when people dont use any form of Xray

### Server Details

This server is running Paper version git-Paper-370 (MC: 1.19.3) (Implementing API version 1.19.3-R0.1-SNAPSHOT) (Git: f88384e)

### Orebfuscator Version

5.3.2

### ProtocolLib Version

5.0.0-SNAPSHOT-b605

### Extra Details

BTW picture provided is taken from a vanilla minecraft launcher no texture packs enabled also not a texture pack on the server

**Comment by Ingrim4** [comment]
Did the player break the blocks with a vanilla pickax or some tool from a plugin/mod that can break blocks for you?

**Comment by Noobymen** [comment]
> Did the player break the blocks with a vanilla pickax or some tool from a plugin/mod that can break blocks for you?

At the time off the screenshot the player was using a vanilla shovel but i got a plugin instald called slimefun but at this time this player did not use any slimefun tools just vanilla shovel

**Comment by Ingrim4** [comment]
Is this a reoccurring problem or something that only happened once? Are their any other plugins that might change block breaking behavior or block drops in any way?

**Comment by Bennyboy2006** [comment]
Same problem here, It is a reoccurring problem, I don't have any plugins that would change block breaking or block drops in any way, yet players are seeing fake ores when not xraying, It they don't seem to appear straight away if someone breaks blocks but once the chunk is loaded and unloaded, there always seems to be some fake ores appearing on them, when you punch the blocks they disappear. Also when you have insta-mine on sculk blocks you see ores appear under the sculk as you mine it for a split second, This isn't much of a problem as you can barely notice it and by the time the sculk is gone the blocks have returned to their proper state

**Comment by Noobymen** [comment]
> Is this a reoccurring problem or something that only happened once? Are their any other plugins that might change block breaking behavior or block drops in any way?

Yes it happends many time and about plugins maybe slimefun will do that yeah

**Comment by Ingrim4** [comment]
Not sure if this is the same problem as in #255. Could you try if this will fix it for you (might tank performance a little bit though): https://github.com/Imprex-Development/Orebfuscator/suites/10133768419/artifacts/494732678

**Comment by Noobymen** [comment]
> Not sure if this is the same problem as in #255. Could you try if this will fix it for you (might tank performance a little bit though): https://github.com/Imprex-Development/Orebfuscator/suites/10133768419/artifacts/494732678

Yeah i tested this and now no random ores pop up anymore so what ever is in this version off the plugin it needs to be implement into the official version off the plugin

**Comment by roracle** [comment]
Hello. I can confirm that using Slimefun's Gold Pan item leaves obfuscated ores visible. This is happening due to the gold pan pretty much removing the gravel block instead of breaking it (so I assume).

**Comment by Noobymen** [comment]
> Hello. I can confirm that using Slimefun's Gold Pan item leaves obfuscated ores visible. This is happening due to the gold pan pretty much removing the gravel block instead of breaking it (so I assume).

Me too just tested it with the modified plugin you give me off orebfuscator random ores pop up when using gold pan from slimefun to get item's from gravel @Ingrim4

**Comment by roracle** [comment]
> > Hello. I can confirm that using Slimefun's Gold Pan item leaves obfuscated ores visible. This is happening due to the gold pan pretty much removing the gravel block instead of breaking it (so I assume).
> 
> Me too just tested it with the modified plugin you give me off orebfuscator random ores pop up when using gold pan from slimefun to get item's from gravel @Ingrim4

Paper has built in obfuscation now. I might be trying that out next week. Our server really doesn't need it though, since SF has the Advanced Industrial Miner, there's almost no point in using xray.

**Comment by Ingrim4** [comment]
> Yeah i tested this and now no random ores pop up anymore so what ever is in this version off the plugin it needs to be implement into the official version off the plugin

I will try to add this to the master branch once I have more time (at the earliest next month).

> Hello. I can confirm that using Slimefun's Gold Pan item leaves obfuscated ores visible. This is happening due to the gold pan pretty much removing the gravel block instead of breaking it (so I assume).

We do expose an API for other plugins that change blocks and SlimeFun once had an update Orebfuscator integration but I guess they didn't add new items to said integration. Nothing we can do about it, papers built-in anti-xray might also have the same issue because I don't know if they trigger block updates if a plugin changed a block.

> Paper has built in obfuscation now. I might be trying that out next week. Our server really doesn't need it though, since SF has the Advanced Industrial Miner, there's almost no point in using xray.

Their are multiple reason why Orebfuscate is still being maintained in spite of papers built-in anti-xray. Here is a short list of them: https://github.com/Imprex-Development/Orebfuscator/wiki/Frequently-Asked-Questions#why-use-orebfuscator-over-papers-built-in-anti-xray

**Comment by NgLoader** [comment]
Related to https://github.com/Slimefun/Slimefun4/issues/3764
